src/rdkafka_partition.c

```diff
@@ -2848,39 +2925,76 @@ static int rd_kafka_topic_partition_cmp (const void *_a, const void *_b,
                 return RD_CMP(a->partition, b->partition);
 }
 
+/** @brief Compare only the topic */
+int rd_kafka_topic_partition_cmp_topic (const void *_a, const void *_b) {
+        const rd_kafka_topic_partition_t *a = _a;
+        const rd_kafka_topic_partition_t *b = _b;
+        return strcmp(a->topic, b->topic);
+}
+
+static int rd_kafka_topic_partition_cmp_opaque (const void *_a, const void *_b,
+                                                void *opaque) {
```

Comment:
why is `opaque` here if it's not used?

Reply:
(it seems like it should be encapsulated more, the detail not exposed as high up).

Reply:
It's a leaky abstraction of qsort_r() which takes an opaque arg.
Otherwise we'd have to jump through some hoops to trampoline a call to a users opaque-less cmp-arator.